hooks/pre/31-upgrade-puppet.rb

```diff
@@ -66,6 +66,29 @@ def fail_and_exit(message)
     Kafo::Helpers.reset_value(param('foreman_proxy', 'puppetca_cmd'))
     Kafo::Helpers.reset_value(param('foreman_proxy', 'puppetrun_cmd'))
     Kafo::Helpers.reset_value(param('foreman_proxy_plugin_pulp', 'puppet_content_dir'))
+
+    Kafo::Helpers.reset_value(param('puppet', 'server_implementation'))
+    Kafo::Helpers.reset_value(param('puppet', 'autosign'))
+    Kafo::Helpers.reset_value(param('puppet', 'client_package'))
+    Kafo::Helpers.reset_value(param('puppet', 'codedir'))
+    Kafo::Helpers.reset_value(param('puppet', 'configtimeout'))
+    Kafo::Helpers.reset_value(param('puppet', 'dir'))
+    Kafo::Helpers.reset_value(param('puppet', 'logdir'))
+    Kafo::Helpers.reset_value(param('puppet', 'rundir'))
+    Kafo::Helpers.reset_value(param('puppet', 'ssldir'))
+    Kafo::Helpers.reset_value(param('puppet', 'vardir'))
+    Kafo::Helpers.reset_value(param('puppet', 'server_common_modules_path'))
+    Kafo::Helpers.reset_value(param('puppet', 'server_default_manifest_path'))
+    Kafo::Helpers.reset_value(param('puppet', 'server_dir'))
+    Kafo::Helpers.reset_value(param('puppet', 'server_envs_dir'))
+    Kafo::Helpers.reset_value(param('puppet', 'server_external_nodes'))
+    Kafo::Helpers.reset_value(param('puppet', 'server_jruby_gem_home'))
+    Kafo::Helpers.reset_value(param('puppet', 'server_package'))
+    Kafo::Helpers.reset_value(param('puppet', 'server_puppetserver_dir'))
+    Kafo::Helpers.reset_value(param('puppet', 'server_puppetserver_vardir'))
+    Kafo::Helpers.reset_value(param('puppet', 'server_puppetserver_version'))
```

Comment:
No strong opinion. Since you have to split up those lines it doesn't become much more readable